### PR TITLE
Adding a graphical user interface for those who don't want to use the text to speech voices with the command line

### PR DESCRIPTION
**Pull request by MitchellProductions**
gui

### CODE REVIEW

**Comment by erogol** [comment]
GREAT PR !!

It'd be nice to include some explanation for using your GUI, like how to run it, where to find it etc. 

did you check the contribution guideline https://github.com/coqui-ai/TTS#contribution-guidelines ? 

Please make sure you followed all the steps. 

right now, your PR has linter errors.

**Comment by erogol** [comment]
Couple of problems:

1. You also need to add the extra libraries you used in requirements.txt
2. coqui-tts-qui.py needs to be renamed coqui_tts_qui.py and move under TTS/bin folder since all the executables are there.

**Comment by MitchellProductions** [comment]
@erogol coqui_tts_py relies on models.json which is in TTS/ instead of TTS/bin

File: coqui-tts-gui.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+def generate():
+    if inputbox.get("1.0","end-1c") == "":
+        messagebox.showerror(message = "TTS will give a division by zero error if the text field is blank.")
+    else:
+        if not os.path.exists('coqui-tts-output'):
+            try:
+                os.makedirs('coqui-tts-output')
+            except OSError as e:
+                if e.errno != errno.EEXIST:
```
Comment (erogol) [inline coqui-tts-gui.py:28]: I guess it should be ```e.errno.EEXIST```

File: coqui-tts-gui.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        row = 14, padx = 10, pady = 12) 
+  
+n = tk.StringVar() 
+ttsmodelbox = ttk.Combobox(window, width = 32, textvariable = n, state = "readonly") 
+# Adding combobox drop down list 
+ttsmodelbox['values'] = ('en/ljspeech/glow-tts',  
+                          'en/ljspeech/tacotron2-DCA', 
+                          'en/ljspeech/speedy-speech-wn', 
+                          'es/mai/tacotron2-DDC', 
+                          'fr/mai/tacotron2-DDC')
```
Comment (erogol) [inline coqui-tts-gui.py:159]: read the models from ```.models.json``` instead of hard code.

File: coqui-tts-gui.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+cudacheckbutton = ttk.Checkbutton(window)
+cudacheckbutton.grid(column = 1, columnspan = 1, row = 15, padx = 10, pady = 12)
+cudacheckbutton.state(['!alternate'])
+cudacheckbutton.state(['!selected'])
+
+# Adding combobox drop down list 
+vocodermodelbox['values'] = ('universal/libri-tts/wavegrad',  
+                          'universal/libri-tts/fullband-melgan', 
+                          'en/ljspeech/multiband-melgan')
```
Comment (erogol) [inline coqui-tts-gui.py:186]: same... from ```.models.json```

File: requirements.txt [hunk 4]
```diff
@@ -24,4 +24,5 @@ pylint==2.5.3
 gdown
 umap-learn
 cython
-pyyaml
\ No newline at end of file
+pyyaml
```
Comment (erogol) [inline requirements.txt:27]: I installed all these to run

```
sudo apt install libcairo2-dev
sudo apt install libglib2.0-dev
sudo apt install libgirepository1.0-dev
pip install playsound
pip install pgi
pip install pycairo
pip install PyGObject
```

**Review by erogol — CHANGES_REQUESTED** [review]

**Comment by erogol** [comment]
@MitchellProductions  What is your run-time OS ? Is it windows or linux?

**Comment by erogol** [comment]
I also got this when I try to export the audio 

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/home/erogol/miniconda3/lib/python3.8/tkinter/__init__.py", line 1883, in __call__
    return self.func(*args)
  File "coqui-tts-gui.py", line 81, in exportaudio
    f = filedialog.asksaveasfile(mode='a', defaultextension=".wav", filetypes = [("Wave files", ".wav")])
  File "/home/erogol/miniconda3/lib/python3.8/tkinter/filedialog.py", line 428, in asksaveasfile
    return open(filename, mode)
PermissionError: [Errno 13] Permission denied: '/home/this_is_an_example.wav'
```

**Comment by MitchellProductions** [comment]
@erogol I'm on Windows

**Comment by erogol** [comment]
instead of relying on CI to see the linter output just do the following as shown in contrib. guidelines.

```
cardboardlinter -n auto
```

**Comment by erogol** [comment]
Things above not resolved, pls don't sign them resolved until you do so.

**Comment by MitchellProductions** [comment]
@erogol There are no errors when I ran the cardboardlinter you showed

**Comment by erogol** [comment]
I don't think it runs correctly if there are no errors. You might need to do some windows specific stuff to make it work.

**Comment by erogol** [comment]
you are booming the CI right now. Please dont use CI for testing your code.  Or I need to close the PR until you fix the issues.

**Comment by MitchellProductions** [comment]
@erogol What should I use instead?

File: coqui-tts-gui.py (hunk 1 above)
Comment (SanjaESC) [reply coqui-tts-gui.py]: this needs `import errno` for it to work

**Comment by erogol** [comment]
setup your local env correctly and use it to fix the issues. CI is not a debug tool. Since you are on win i cant be much help but you can ask on chat room or just google.

File: coqui-tts-gui.py (hunk 2 above)
Comment (MitchellProductions) [reply coqui-tts-gui.py]: How can I? https://stackoverflow.com/questions/59760156/how-to-append-values-from-json-file-to-tkinter-combobox-in-python

**Comment by erogol** [comment]
I close this until you fix the issues in your implementation to prevent CI run.

**Pull request closed without merging**